### PR TITLE
Allow compression at the transport level.

### PR DESCRIPTION
**Pull request by ehiggs**
Solves #541

### CODE REVIEW

**Comment by DarthSim** [comment]
LGTM, but there are a few things I'd like to change:
- I don't think that rewriting the response body is a good idea.
- We don't know the actual content length as the content is gzipped. We should pass zero content length in the case of gzip.

Let's change it a little bit like this:

```go
body := res.Body
contentLength := int(res.ContentLength)

if res.Header.Get("Content-Encoding") == "gzip" {
  // ...
  body = gzipBody
  contentLength = 0
}

imgdata, err := readAndCheckImage(body, contentLength)
```

File: download.go [hunk 1]
```diff
@@ -200,6 +201,15 @@ func downloadImage(ctx context.Context) (context.Context, context.CancelFunc, er
 		return ctx, func() {}, err
 	}
 
+	if res.Header.Get("Content-Encoding") == "gzip" {
+		gzipBody, errGzip := gzip.NewReader(res.Body)
+		if errGzip != nil {
+			return ctx, func() {}, err
```
Comment (DarthSim) [inline download.go:207]:
```suggestion
			return ctx, func() {}, errGzip
```

Comment (ehiggs) [reply download.go]: Good catch. I changed the name of the err variable to satisfy the linter.

**Comment by DarthSim** [comment]
LGTM now 👍 Thanks for the contribution!

**Pull request merged**